cdap-gateway/src/test/java/co/cask/cdap/gateway/handlers/metrics/MetricsHandlerTestRun.java

```diff
@@ -92,42 +95,69 @@ private static void setupMetrics() throws Exception {
 
   @Test
   public void testSearchContext() throws Exception {
-    verifySearchResult("/v3/metrics/search?target=childContext&context=WordCount1.f",
-                       ImmutableList.<String>of("WordCounter"));
-    verifySearchResult("/v3/metrics/search?target=childContext&context=WCount1",
-                       ImmutableList.<String>of("b", "f", "p"));
-    verifySearchResult("/v3/metrics/search?target=childContext&context=WCount1.b.ClassicWordCount",
+    // WordCount is in myspace, WCount in yourspace
+    verifySearchResult(getContextSearchUri(MY_SPACE, null), ImmutableList.<String>of("WordCount1"));
+    verifySearchResult(getContextSearchUri(YOUR_SPACE, null), ImmutableList.<String>of("WCount1"));
+
+    // WordCount should be found in myspace, not in yourspace
+    verifySearchResult(getContextSearchUri(MY_SPACE, "WordCount1.f"), ImmutableList.<String>of("WordCounter"));
+    verifySearchResult(getContextSearchUri(YOUR_SPACE, "WordCount1.f"), ImmutableList.<String>of());
+
+    // WCount should be found in yourspace, not in myspace
+    verifySearchResult(getContextSearchUri(YOUR_SPACE, "WCount1"), ImmutableList.<String>of("b", "f", "p"));
+    verifySearchResult(getContextSearchUri(MY_SPACE, "WCount1"), ImmutableList.<String>of());
+
+    // verify other metrics for WCount app
+    verifySearchResult(getContextSearchUri(YOUR_SPACE, "WCount1.b.ClassicWordCount"),
                        ImmutableList.<String>of("m", "r"));
-    verifySearchResult("/v3/metrics/search?target=childContext&context=WCount1.b.ClassicWordCount.m",
-                       ImmutableList.<String>of());
+    verifySearchResult(getContextSearchUri(YOUR_SPACE, "WCount1.b.ClassicWordCount.m"), ImmutableList.<String>of());
   }
 
   @Test
   public void testQueryMetrics() throws Exception {
-    // aggregate result
+    // aggregate result, in the right namespace
     verifyAggregateQueryResult(
-      "/v3/metrics/query?context=WCount1.f.WCounter.splitter&metric=system.reads&aggregate=true", 3);
+      getMetricsAggregateQueryUri(YOUR_SPACE, "WCount1.f.WCounter.splitter", "system.reads"), 3);
     verifyAggregateQueryResult(
-      "/v3/metrics/query?context=WCount1.f.WCounter.counter&metric=system.reads&aggregate=true", 1);
+      getMetricsAggregateQueryUri(YOUR_SPACE, "WCount1.f.WCounter.counter", "system.reads"), 1);
+
+    // aggregate result, in the wrong namespace
+    verifyAggregateQueryResult(getMetricsAggregateQueryUri(MY_SPACE, "WCount1.f.WCounter.splitter", "system.reads"), 0);
 
     // time range
     // now-60s, now+60s
-    verifyRangeQueryResult(
-      "/v3/metrics/query?context=WCount1.f.WCounter.splitter&metric=system.reads&start=now%2D60s&end=now%2B60s", 2, 3);
+    verifyRangeQueryResult(getMetricsRangeQueryUri(YOUR_SPACE, "WCount1.f.WCounter.splitter", "system.reads",
```

Comment:
I thought they make it easier to write more such queries in future. Again, don't feel very strongly about it. Had just done it so as to develop newer unit tests. I'll make it the way it was.